### PR TITLE
ui: add home page

### PR DESCRIPTION
**Pull request by Joe2k**
For #67 

- Moved current home page to dashboard which can be viewed only when logged in
- Created new home page which will be like the landing page.
- Added features(Whatever I know) and contributors. 

Please feel free to suggest any edit and addition of stuff here as even I'm experimenting!

Adding screenshots here.
Screenshot 2023-08-30 at 2 18 29 AM
Screenshot 2023-08-30 at 2 18 49 AM

### CODE REVIEW

**Comment by Joe2k** [comment]
@punchagan I'm not sure why the Integration tests are failing. I changed the base_url in `test_ui` from `/` to `/login` as now we don't navigate to login page from the home screen. But that also gives 404

File: frontend/src/components/Dashboard.js [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        data-active-classes="bg-white dark:bg-gray-900 text-gray-900 dark:text-white"
+        data-inactive-classes="text-gray-500 dark:text-gray-400"
+      >
+        <Show when={store.data.player}>
+          <h2 id="accordion-heading-player">
+            <button
+              type="button"
+              class="flex items-center justify-between w-full py-5 font-medium text-left text-gray-500 border-b border-gray-200 dark:border-gray-700 dark:text-gray-400"
+              data-accordion-target="#accordion-body-player"
+              aria-expanded="false"
```
Comment (punchagan) [inline frontend/src/components/Dashboard.js:107]: Is there a reason you changed this from what was implemented in 2189b8b14eb74d59e896aeacf8393a819c966200. 
I implemented this to let see a "call to action" if their profile is incomplete. But, may be we can also show the status in the accordion header itself. What was your reasoning?

File: frontend/src/components/Dashboard.js [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+              </>
+            )}
+          </For>
+        </Show>
+        <h2 id="accordion-heading-actions">
+          <button
+            type="button"
+            class="flex items-center justify-between w-full py-5 font-medium text-left text-gray-500 border-b border-gray-200 dark:border-gray-700 dark:text-gray-400"
+            data-accordion-target="#accordion-body-actions"
+            aria-expanded="false"
```
Comment (punchagan) [inline frontend/src/components/Dashboard.js:158]: Expanded user actions accordion if user hasn't done any registration stuff was implemented in abff00026592d3c4503a1936246fc06e681c694d. Again, not sure why this was changed. :)

File: frontend/src/components/Dashboard.js [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+        <Show when={store.data.wards}>
+          <For each={store.data.wards}>
+            {ward => (
+              <>
+                <h2 id={`accordion-heading-ward-${ward.id}`}>
+                  <button
+                    type="button"
+                    class="flex items-center justify-between w-full py-5 font-medium text-left text-gray-500 border-b border-gray-200 dark:border-gray-700 dark:text-gray-400"
+                    data-accordion-target={`#accordion-body-ward-${ward.id}`}
+                    aria-expanded="false"
```
Comment (punchagan) [inline frontend/src/components/Dashboard.js:133]: fcc861224accd464b638ea9dcc2a9825aaed68dd also expands the wards accordions "if required"

File: frontend/src/components/Home.js [hunk 4]
```diff
@@ -1,223 +1,120 @@
...
+                  {feature.description}
+                </p>
+              </div>
+            )}
+          </For>
+        </div>
+      </div>
+      <div class="mt-10">
+        <h3 class="text-2xl font-extrabold text-center">
+          <mark class="px-2 text-white bg-blue-600 rounded dark:bg-blue-500">
```
Comment (punchagan) [inline frontend/src/components/Home.js:96]: This is great. Can we link to the repository, here?

**Review by punchagan — COMMENTED** [review]
This looks great! Thanks @Joe2k. I had a few questions and suggestions, otherwise it looks great! :tada:

**Comment by punchagan** [comment]
> @punchagan I'm not sure why the Integration tests are failing. I changed the base_url in `test_ui` from `/` to `/login` as now we don't navigate to login page from the home screen. But that also gives 404

I can fix the integration tests for you... once you address the review comments.

File: frontend/src/components/Dashboard.js (hunk 1 above)
Comment (Joe2k) [reply frontend/src/components/Dashboard.js:107]: Ahh I think something went wrong when I rebased from newer commits.. I didnt change anything in Dashboard except the file name change xD

File: frontend/src/components/Dashboard.js (hunk 2 above)
Comment (Joe2k) [reply frontend/src/components/Dashboard.js:158]: Same as [here](https://github.com/india-ultimate/hub/pull/75#discussion_r1309637177)

File: frontend/src/components/Dashboard.js (hunk 3 above)
Comment (Joe2k) [reply frontend/src/components/Dashboard.js:133]: Same as [here](https://github.com/india-ultimate/hub/pull/75#discussion_r1309637177)

File: frontend/src/components/Home.js (hunk 4 above)
Comment (Joe2k) [reply frontend/src/components/Home.js:96]: Yep Let me do that

File: frontend/src/components/Dashboard.js (hunk 1 above)
Comment (Joe2k) [reply frontend/src/components/Dashboard.js:107]: Lemme change to latest code and push just for Dashboard.js

Comment (punchagan) [reply frontend/src/components/Dashboard.js:107]: Are you going to fix the rebase? Or do you want me to do it? :)

Comment (Joe2k) [reply frontend/src/components/Dashboard.js:107]: So I see that only Dashboard.js is not upto date. So I'm copy pasting from the main brach Home.js code here and pushing it. That'll work right?

Comment (punchagan) [reply frontend/src/components/Dashboard.js:107]: Yes, that should work! :ok_hand:

File: frontend/src/components/Home.js (hunk 4 above)
Comment (Joe2k) [reply frontend/src/components/Home.js]: Linked it now

File: frontend/src/components/Dashboard.js (hunk 1 above)
Comment (Joe2k) [reply frontend/src/components/Dashboard.js]: Pushed it

**Comment by punchagan** [comment]
> @punchagan I'm not sure why the Integration tests are failing. I changed the base_url in `test_ui` from `/` to `/login` as now we don't navigate to login page from the home screen. But that also gives 404

Navigating to the `/login` on the Django server doesn't work: the `base_url` was referring to the Django server. We could directly navigate to the `/login` on the webpack server, but then CSRF cookies are not set, and the API calls fail. So, I changed the tests to navigate to the `base_url` as before, but click on the Login button, before proceeding with the rest of the test.

**Comment by punchagan** [comment]
This is great! Thanks for working on this, @Joe2k ! :tada: 

Merged the changes manually as https://github.com/india-ultimate/hub/commit/40dd1fd

**Pull request closed without merging**